src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageRectangleSelection/ImageRectangleSelection.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import React from 'react';
+import RectangleSelection from '../RectangleSelection';
+import RoundingNormalizer from '../RectangleSelection/dataNormalizers/RoundingNormalizer';
+import type {SelectionData} from '../RectangleSelection/types';
+import containerSizeAware from '../containerSizeAware';
+import log from 'loglevel';
+import {observer} from 'mobx-react';
+import styles from './imageRectangleSelection.scss';
+
+@observer
+export class ImageRectangleSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+        containerWidth: number,
+        containerHeight: number,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+    @observable imageLoaded = false;
+
+    naturalHorizontalToReal = (h: number) => h * this.imageResizedWidth / this.image.naturalWidth;
+    scaledHorizontalToNatural = (h: number) => h * this.image.naturalWidth / this.imageResizedWidth;
+    naturalVerticalToReal = (v: number) => v * this.imageResizedHeight / this.image.naturalHeight;
+    scaledVerticalToNatural = (v: number) => v * this.image.naturalHeight / this.imageResizedHeight;
+
+    naturalDataToReal(data: SelectionData): SelectionData {
+        return {
+            width: this.naturalHorizontalToReal(data.width),
+            height: this.naturalVerticalToReal(data.height),
+            left: this.naturalHorizontalToReal(data.left),
+            top: this.naturalVerticalToReal(data.top),
+        };
+    }
+
+    scaledDataToNatural(data: SelectionData): SelectionData {
+        return {
+            width: this.scaledHorizontalToNatural(data.width),
+            height: this.scaledVerticalToNatural(data.height),
+            left: this.scaledHorizontalToNatural(data.left),
+            top: this.scaledVerticalToNatural(data.top),
+        };
+    }
+
+    componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image "' + this.props.src + '"');
```

Comment:
I don't think the error could be handled as the onerror gets executed asynchronously.